cpu/stm32f1/periph/spi.c

```diff
@@ -210,27 +211,35 @@ int spi_transfer_byte(spi_t dev, char out, char *in)
             return -1;
     }
 
-    while (!(spi->SR & SPI_SR_TXE)) {}
-    spi->DR = out;
-    transferred++;
-
-    while (!(spi->SR & SPI_SR_RXNE)) {}
-    if (in != NULL) {
-        *in = spi->DR;
-        transferred++;
-    }
-    else {
+    if(!in){
+        for (unsigned i = 0; i < length; i++) {
+            while (!(spi->SR & SPI_SR_TXE)) {}
+            spi->DR = (uint8_t)out[i];
+            transferred++;
+        }
+        /* SPI busy */
+        while ((spi->SR & SPI_SR_BSY)) {}
         spi->DR;
     }
+    else{
+        char tmp;
+        for (unsigned i = 0; i < length; i++) {
+            tmp = (out) ? out[i] : 0;
+            while (!(spi->SR & SPI_SR_TXE)) {}
+            spi->DR = tmp;
+            transferred++;
+            while (!(spi->SR & SPI_SR_RXNE)) {}
+            in[i] = (char)spi->DR;
+            transferred++;
```

Comment:
Is it desired to increase `transferred` twice when you are sending and receiving? If not, you could use `transferred` (or a renamed version of it) as a counter for the loops.